test/helpers/utils.go

```diff
@@ -1107,3 +1132,25 @@ func DeleteNodeAndMachine(t *testing.T, cs *framework.ClientSet, node corev1.Nod
 
 	require.NoError(t, delErr)
 }
+
+// AssertFeatureGatesEnabled fatally exits the test if any feature gate in requiredFeatureGates is not enabled
+func AssertFeatureGatesEnabled(requiredFeatureGates ...string) {
+	cs := framework.NewClientSet("")
+	currentFeatureGates, err := cs.ConfigV1Interface.FeatureGates().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		log.Fatalf("Failed to fetch feature gates")
+	}
+	for _, featureGateName := range requiredFeatureGates {
```

Comment:
suggestion (non-blocking): You can use the sets package in apimachinery to simplify this a bit and return a useful error message whenever a missing FeatureGate is found.

Here's an example you can crib from:

```golang
package main

import (
	"context"
	"fmt"
	"log"

	configv1 "github.com/openshift/api/config/v1"
	"github.com/openshift/machine-config-operator/test/framework"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/apimachinery/pkg/util/sets"
)

// Generally speaking, if a Golang function can cause a program to exit
// prematurely (e.g., calling log.Fatal() or panic()), Golang convention is for
// it to have the word "Must" at the beginning. This is not required, but is
// generally a good idea.
func MustHaveFeatureGatesEnabled(requiredFeatureGates ...configv1.FeatureGateName) {
	cs := framework.NewClientSet("")

	if err := validateFeatureGatesEnabled(cs, requiredFeatureGates...); err != nil {
		log.Fatalln(err)
	}

	log.Printf("All required featuregates %v present!", requiredFeatureGates)
}

// We can simplify detecting enabled feature gate names by using the sets
// package within apimachinery (under k8s.io/apimachinery/pkg/util/sets).
func validateFeatureGatesEnabled(cs *framework.ClientSet, requiredFeatureGates ...configv1.FeatureGateName) error {
	currentFeatureGates, err := cs.ConfigV1Interface.FeatureGates().Get(context.TODO(), "cluster", metav1.GetOptions{})
	if err != nil {
		return fmt.Errorf("failed to fetch feature gates: %w", err)
	}

	// This uses the new Go generics to construct a typed set of
	// FeatureGateNames. Under the hood, sets are map[T]struct{}{} where
	// only the keys matter and one cannot have duplicate keys. Perfect for our use-case!
	enabledFeatures := sets.New[configv1.FeatureGateName]()

	// Load all of the feature gate names into our set. Duplicates will be
	// automatically be removed.
	for _, currentFeatureGateDetails := range currentFeatureGates.Status.FeatureGates {
		for _, enabled := range currentFeatureGateDetails.Enabled {
			enabledFeatures.Insert(enabled.Name)
		}
	}

	// If we have all of the required feature gates, we're done!
	if enabledFeatures.HasAll(requiredFeatureGates...) {
		return nil
	}

	// If we don't, lets diff against what we have vs. what we want and return that information.
	requiredFeatures := sets.New[configv1.FeatureGateName](requiredFeatureGates...)
	disabledRequiredFeatures := requiredFeatures.Difference(enabledFeatures)
	return fmt.Errorf("missing required FeatureGate(s): %v, have: %v", sets.List(disabledRequiredFeatures), sets.List(enabledFeatures))
}

func main() {
	// These string literals will be implicitly cast to configv1.FeatureGateName
	// since their underlying type is a string.
	MustHaveFeatureGatesEnabled("OnClusterBuilds")
}
```

It is completely up to you whether you choose to adopt this or not :smile:

Reply:
This is neat! Will add this in, thanks!